packages/api/src/lib/structures/RouteStore.ts

```diff
@@ -34,18 +36,41 @@ export class RouteStore extends Store<Route> {
 		const methodTable = this.table.get(method as Methods);
 		if (typeof methodTable === 'undefined') return null;
 
-		const parsed = new URL(request.url ?? '');
-		const splitUrl = parsed.pathname.split('/');
+		const { splits, querystring } = this.parseURL(request.url);
+
 		for (const [route, cb] of methodTable.entries()) {
-			const result = route.router.match(splitUrl);
+			const result = route.router.match(splits);
 			if (result === null) continue;
 
 			request.params = result;
-			request.query = Object.fromEntries(parsed.searchParams.entries());
+			request.query = Object.fromEntries(new URLSearchParams(querystring).entries());
 
 			return { route, cb };
 		}
 
 		return null;
 	}
+
+	private parseURL(url = '') {
+		const index = url.indexOf('?');
+
+		/* eslint-disable @typescript-eslint/init-declarations */
+		let pathname: string;
+		let querystring: string;
+		/* eslint-enable @typescript-eslint/init-declarations */
```

Comment:
```suggestion
		let pathname = '';
		let querystring = '';
```

Midnight Kyra pls

Reply:
How about we don't do stupid assignments that serve no purpuse but to waste memory and CPU needlessly? I never liked that rule, and I ignore it 99% of the time because it adds a lot of clutter, in other words, it does more harm than good. :)

Reply:
Sounds like 2018 Kyra